cmp/options.go

```diff
@@ -183,9 +183,17 @@ func Ignore() Option {
 // The transformer f must be a function "func(T) R" that converts values of
 // type T to those of type R and is implicitly filtered to input values
 // assignable to T. The transformer must not mutate T in any way.
-// If T and R are the same type, an additional filter must be applied to
-// act as the base case to prevent an infinite recursion applying the same
-// transform to itself (see the SortedSlice example).
+//
+// To help prevent infinite cycles recursively applying the same transform to
+// the output of itself (e.g., in the case where the input and output types
+// are the same), an implicit filter is added such that a transformer is
+// applicable only if that exact transformer is not already in the
+// tail of the Path since the last non-Transform step.
+// While this is sufficient for most usages of Transformers, this does not
```

Comment:
Are you concerned about how it is documented here? Or are you saying that the implementation is not equivalent to what you had in mind?

If the former, we can adjust the prose written.
If the later, what makes this current implementation different? I'm just trying to go for a minimal implementation so that we can test out ideas.